src/software/simulated_tests/non_terminating_validation_functions/ball_always_moves_forward_validation.cpp

```diff
@@ -0,0 +1,23 @@
+#include "software/simulated_tests/non_terminating_validation_functions/ball_always_moves_forward_validation.h"
+
+void ballAlwaysMovesForward(std::shared_ptr<World> world_ptr,
+                            ValidationCoroutine::push_type& yield)
+{
+    // scored is declared static because we need to remember that we scored if the ball
+    // appears to go past the enemy goal once scored and this terminating function
+    // continues to pass
+    static bool scored = false;
+    if (!scored)
+    {
+        scored = contains(world_ptr->field().enemyGoal(), world_ptr->ball().position());
+    }
```

Comment:
Why does this need to be static? Could you just do

```
if(contains(world_ptr->field().enemyGoal(), world_ptr->ball().position())
{
    return;
}
```?

Reply:
discussed and fixed